buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -409,9 +409,12 @@ class BuildPlugin implements Plugin<Project> {
         project.afterEvaluate {
             // fail on all javac warnings
             project.tasks.withType(JavaCompile) {
-                options.fork = true
-                options.forkOptions.executable = new File(project.javaHome, 'bin/javac')
-                options.forkOptions.memoryMaximumSize = "1g"
+                File gradleJavaHome = Jvm.current().javaHome
+                if (new File(project.javaHome).canonicalPath != gradleJavaHome.canonicalPath) {
+                    options.fork = true
```

Comment:
I get that we used to have to fork for java 9 all because gradle ran in java 8, but I think we can still fork if we want to. Do we no longer want to fork?